### PR TITLE
[SYCL][ESIMD]Limit bfloat16 operators to scalars to enable operations with simd vectors

### PR DESCRIPTION
**Pull request by fineg74**
The purpose of this change is to limit operators defined for bfloat16 to scalar types to allow arithmetic operations between bfloat16 scalars and simd vectors. This allows to use simd operators that are defined separately and support operations between vectors and scalars

### CODE REVIEW

File: sycl/include/sycl/ext/intel/esimd/detail/bfloat16_type_traits.hpp [hunk 1]
```diff
@@ -104,6 +104,8 @@ inline std::ostream &operator<<(std::ostream &O, bfloat16 const &rhs) {
   return O;
 }
 
+template <> struct is_esimd_arithmetic_type<bfloat16, void> : std::true_type {};
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/esimd/detail/bfloat16_type_traits.hpp:107]: Is it something what became unused in process of developing the patch?
I don't see any uses of this struct.

File: sycl/test-e2e/ESIMD/regression/bfloat16_vector_plus_scalar.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  std::cout << "Testing case=" << Case << std::endl;
+
+  constexpr int N = 8;
+  T *Mem = malloc_shared<T>(N, Q);
+  T TOne = static_cast<T>(1);
+  T TTen = static_cast<T>(10);
+
+  Q.single_task([=]() SYCL_ESIMD_KERNEL {
+     simd<T, N> Vec(TOne);
+     Vec = Vec + TTen;
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/regression/bfloat16_vector_plus_scalar.cpp:28]: Please test other operations too. `+`, `-`, `*`, `+=`, etc.

File: sycl/include/sycl/ext/intel/esimd/detail/bfloat16_type_traits.hpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/include/sycl/ext/intel/esimd/detail/bfloat16_type_traits.hpp:97]: Right, your comment helped me to refresh that in memory. Thank you.

**Review by bso-intel — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]
ESIMD changes look good to me.

**Pull request merged**